presto-main/src/main/java/io/prestosql/execution/CreateTableTask.java

```diff
@@ -170,6 +174,26 @@ else if (element instanceof LikeClause) {
                     inheritedProperties = likeTableMetadata.getMetadata().getProperties();
                 }
 
+                if (propertiesOption.orElse(EXCLUDING) == INCLUDING) {
+                    try {
+                        accessControl.checkCanShowCreateTable(session.toSecurityContext(), likeTableName);
+                    }
+                    catch (AccessDeniedException e) {
+                        throw new AccessDeniedException("Cannot reference properties of table " + likeTableName);
+                    }
+                }
+                else {
+                    try {
+                        accessControl.checkCanSelectFromColumns(
```

Comment:
I think we should check this unconditionally (always). Successful `checkCanShowCreateTable` does not imply successful `checkCanSelectFromColumns`.